test/unit/serverless/aws-lambda.test.js

```diff
@@ -94,39 +93,43 @@ describe('AwsLambda.patchLambdaHandler', () => {
     awsLambda = null
   })
 
-  it('should return original handler if not a function', () => {
+  t.test('should return original handler if not a function', (t) => {
     const handler = {}
     const newHandler = awsLambda.patchLambdaHandler(handler)
 
-    expect(newHandler).to.equal(handler)
+    t.equal(newHandler, handler)
+    t.end()
   })
 
-  it('should pick up on the arn', function () {
-    expect(agent.collector.metadata.arn).to.be.null
+  t.test('should pick up on the arn', function (t) {
+    t.equal(agent.collector.metadata.arn, null)
     awsLambda.patchLambdaHandler(() => {})(stubEvent, stubContext, stubCallback)
-    expect(agent.collector.metadata.arn).to.equal(stubContext.invokedFunctionArn)
+    t.equal(agent.collector.metadata.arn, stubContext.invokedFunctionArn)
+    t.end()
   })
 
-  describe('when invoked with API Gateway Lambda proxy event', () => {
+  t.test('when invoked with API Gateway Lambda proxy event', (t) => {
+    t.autoend()
+
     const validResponse = {
       isBase64Encoded: false,
       statusCode: 200,
       headers: { responseHeader: 'headerValue' },
       body: 'worked'
     }
 
-    it('should create web transaction', (done) => {
+    t.test('should create web transaction', async (t) => {
```

Comment:
I don't see this test doing anything promise based, should drop the `async`.

Reply:
Would it be alright if I merged now but followed up later by including https://eslint.org/docs/rules/require-await as part of our linter ruleset?

Reply:
Yeah, that works for me. And that seems like a good rule to have. For the most part, we'll naturally avoid production issues because we are writing code that supports both but always good to be in the habit of avoiding unnecessary promises.

Reply:
FWIW when this rule is added, it should be added to our [central ruleset](https://github.com/newrelic/eslint-config-newrelic) and not just in this agent as we'd want this across all repos

Reply:
Made an issue about it here: https://github.com/newrelic/eslint-config-newrelic/issues/5